Source/JavaScriptCore/b3/air/AirAllocateRegistersAndStackAndGenerateCode.cpp

```diff
@@ -143,19 +143,39 @@ void GenerateAndAllocateRegisters::insertBlocksForFlushAfterTerminalPatchpoints(
     blockInsertionSet.execute();
 }
 
-static ALWAYS_INLINE CCallHelpers::Address callFrameAddr(Air::Opcode opcode, CCallHelpers& jit, intptr_t offsetFromFP)
-{
+static ALWAYS_INLINE Air::Arg callFrameAddr(Air::Opcode opcode, CCallHelpers& jit, intptr_t offsetFromFP, intptr_t frameSize, Width width)
+{    
     if (isX86()) {
         ASSERT(Arg::addr(Air::Tmp(GPRInfo::callFrameRegister), offsetFromFP).isValidForm(Move, Width64));
     }
 
     auto addr = Arg::addr(Air::Tmp(GPRInfo::callFrameRegister), offsetFromFP);
-    if (addr.isValidForm(opcode, registerWidth()))
-        return CCallHelpers::Address(GPRInfo::callFrameRegister, offsetFromFP);
+    if (addr.isValidForm(opcode, width))
+        return addr;
+
+    if (isARM64() || isARM_THUMB2()) {
```

Comment:
We should not have arch condition here. If this is only allowed for these archs, then isValidForm should say false to wrong archs instead.

Reply:
I removed those in https://github.com/WebKit/WebKit/commit/da11283e6bd1aaa4c73f84c839f1a7b72f508ef3